gpcontrib/arenadata_toolkit/src/tf_shmem.c

```diff
@@ -0,0 +1,175 @@
+#include "bloom_set.h"
+#include "tf_shmem.h"
+
+#include "storage/ipc.h"
+#include "storage/shmem.h"
+
+#include "arenadata_toolkit_guc.h"
+
+#include <math.h>
+
+static shmem_startup_hook_type next_shmem_startup_hook = NULL;
+tf_shared_state_t *tf_shared_state = NULL;
+LWLock *bloom_set_lock;
+tf_entry_lock_t bloom_locks[MAX_DB_TRACK_COUNT];
+uint64 bloom_hash_seed;
+int bloom_hash_num;
+
+/*
+ * Separate initialization of LWLocks;
+ */
+static void
+init_lwlocks(void)
+{
+	bloom_set_lock = LWLockAssign();
```

Comment:
We should assign LWLocks only once when `ShmemInitStruct()` initially was called, like it is done in the [sample](https://github.com/arenadata/gpdb/blob/eeee4d43b7179a69cd7ea4234736d214d4dffaf7/contrib/pg_stat_statements/pg_stat_statements.c#L439)

Reply:
When shmem_startup_hook is called more than once?